spring-boot-project/spring-boot/src/main/java/org/springframework/boot/webservices/client/WebServiceTemplateBuilder.java

```diff
@@ -0,0 +1,1106 @@
+			Class<? extends WebServiceMessageSender> webServiceMessageSenderClass) {
+		Assert.notNull(webServiceMessageSenderClass,
+				"webServiceMessageSenderClass must not be null");
+		return addWebServiceMessageSender(
+				supplier(webServiceMessageSenderClass, BeanUtils::instantiateClass));
+	}
+
+	/**
+	 * Sets the message factory used for creating messages.
+	 * @param messageFactory instance of WebServiceMessageFactory
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setMessageFactory(WebServiceMessageFactory)
+	 **/
+
+	public WebServiceTemplateBuilder setWebServiceMessageFactory(
+			WebServiceMessageFactory messageFactory) {
+		Assert.notNull(messageFactory, "messageFactory must not be null");
+		return new WebServiceTemplateBuilder(this.interceptors, this.internalCustomizers,
+				this.customizers, this.webServiceMessageSenderSuppliers,
+				this.webServiceMessageSenderCustomizers, this.marshaller,
+				this.unmarshaller, this.destinationProvider, this.transformerFactoryClass,
+				messageFactory, this.detectWebServiceMessageSender);
+	}
+
+	/**
+	 * Set {@link WebServiceTemplate#setUnmarshaller(Unmarshaller) unmarshaller} on the
+	 * underlying.
+	 * @param unmarshaller message unmarshaller
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setUnmarshaller(Unmarshaller)
+	 **/
+	public WebServiceTemplateBuilder setUnmarshaller(Unmarshaller unmarshaller) {
+		Assert.notNull(unmarshaller, "unmarshaller must not be null");
+		return new WebServiceTemplateBuilder(this.interceptors, this.internalCustomizers,
+				this.customizers, this.webServiceMessageSenderSuppliers,
+				this.webServiceMessageSenderCustomizers, this.marshaller, unmarshaller,
+				this.destinationProvider, this.transformerFactoryClass,
+				this.messageFactory, this.detectWebServiceMessageSender);
+	}
+
+	/**
+	 * Set {@link WebServiceTemplate#setMarshaller(Marshaller) marshaller} on the
+	 * underlying.
+	 * @param marshaller message marshaller
+	 * @return a new builder instance.
+	 * @see WebServiceTemplate#setMarshaller(Marshaller)
+	 **/
+	public WebServiceTemplateBuilder setMarshaller(Marshaller marshaller) {
+		Assert.notNull(marshaller, "marshaller must not be null");
+		return new WebServiceTemplateBuilder(this.interceptors, this.internalCustomizers,
+				this.customizers, this.webServiceMessageSenderSuppliers,
+				this.webServiceMessageSenderCustomizers, marshaller, this.unmarshaller,
+				this.destinationProvider, this.transformerFactoryClass,
+				this.messageFactory, this.detectWebServiceMessageSender);
+	}
+
+	/**
+	 * Sets the connection timeout in milliseconds on the underlying.
+	 * @param connectionTimeout the connection timeout in milliseconds
+	 * @return a new builder instance.
+	 * @throws java.lang.IllegalStateException if the underlying source doesn't support a
+	 * connection timeout.
+	 */
+	public WebServiceTemplateBuilder setConnectionTimeout(int connectionTimeout) {
```

Comment:
Considering that we may have multiple `WebServiceMessageSender`, I don't think we should expose this method here. You can provide a configured `WebServiceMessageSender`, this feels weird to me that all of them are reconfigured behind the scenes.